### ISSUE TITLE
Having client name in SubVersion fails a make check test

### ISSUE THREAD

**Issue by UncleHunto** [UASF/bitcoin#4]
### Describe the issue

In the 0.14 branch, from what I can tell, the UASF version part of the client name, looks like it is currently located in the "SubVersion" section. Having it located here appears to cause a "make check"
test to fail, if you run it after building from source. I noticed that this did not occur before commit 8bada77 "Update client name". In this commit, the UASF version appears to be moved from the primary version name, to the subversion. When the UASF version was in the primary version location, it passed all "make check" tests. I'm not sure if this is really an important issue or not, forgive me if this doesn't matter, but I thought I would mention it just in case.

#### If so, please list the steps to reproduce below:
1.  Build from 0.14 branch
2.  Run "make check"

### Expected behaviour
Pass all tests

### Actual behaviour
Failed one test

### What version of bitcoin-core are you using?
branch 0.14 from this repo

### Machine specs:
- OS: Manjaro/Arch
- CPU: Intel i7
- RAM: 32 GB
- Disk size: 8 TB
- Disk Type (HD/SDD): HD

### Any extra information that might be useful in the debugging process.
the relevant log file contents:
```
=============================================
   Bitcoin Core 0.14.0: src/test-suite.log
=============================================

# TOTAL: 2
# PASS:  1
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: test/test_bitcoin
=======================

Running 228 test cases...
test/util_tests.cpp(499): error: in "util_tests/test_FormatSubVersion": check FormatSubVersion("Test", 99900, std::vector<std::string>()) == std::string("/Test:0.9.99/") has failed [/Test:0.9.99/UASF-Segwit:0/ != /Test:0.9.99/]
test/util_tests.cpp(500): error: in "util_tests/test_FormatSubVersion": check FormatSubVersion("Test", 99900, comments) == std::string("/Test:0.9.99(comment1)/") has failed [/Test:0.9.99(comment1)/UASF-Segwit:0/ != /Test:0.9.99(comment1)/]
test/util_tests.cpp(501): error: in "util_tests/test_FormatSubVersion": check FormatSubVersion("Test", 99900, comments2) == std::string("/Test:0.9.99(comment1; Comment2; .,_?@-; )/") has failed [/Test:0.9.99(comment1; Comment2; .,_?@-; )/UASF-Segwit:0/ != /Test:0.9.99(comment1; Comment2; .,_?@-; )/]

*** 3 failures are detected in the test module "Bitcoin Test Suite"
FAIL test/test_bitcoin (exit status: 201)

```

**Comment by sneurlax** [comment]
I haven't actually tried building 0.14.0-UASF yet, so I'll go ahead and try and confirm or deny this for my system

**Comment by UASF** [comment]
See [PR#6](https://github.com/UASF/bitcoin/pull/6). However not sure this is the correct way to fix this. Waiting for comments here.

**Comment by UncleHunto** [comment]
I tested out pull request #6 and it does fix this issue. I'm not sure if this is the "correct" way to fix it, either, but it does the job.

**Comment by UASF** [comment]
Fixed by  #7  closing.
